src/main/java/minijava/ir/optimize/JmpBlockRemover.java

```diff
@@ -0,0 +1,63 @@
+package minijava.ir.optimize;
+
+import static org.jooq.lambda.Seq.seq;
+
+import com.google.common.collect.Iterables;
+import firm.BackEdges;
+import firm.Graph;
+import firm.Mode;
+import firm.nodes.Block;
+import firm.nodes.Jmp;
+import firm.nodes.Node;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.StreamSupport;
+
+/** Removes {@link Block} nodes that contain a single {@link Jmp} node and nothing else. */
+public class JmpBlockRemover extends BaseOptimizer {
+
+  @Override
+  public boolean optimize(Graph graph) {
+    this.graph = graph;
+    BackEdges.enable(graph);
+    graph.walkTopological(this);
+    BackEdges.disable(graph);
+    return hasChanged;
+  }
+
+  @Override
+  public void visit(Block block) {
+    if (!hasChanged && isJmpBlock(block)) {
+      hasChanged = true;
+      remove(block);
+    }
+  }
+
+  private boolean isJmpBlock(Block block) {
+    List<Node> nodesInBlock = nodesInBlock(block);
+    return nodesInBlock.size() == 1 && nodesInBlock.get(0) instanceof Jmp;
+  }
+
+  private List<Node> nodesInBlock(Block block) {
+    return StreamSupport.stream(BackEdges.getOuts(block).spliterator(), false)
```

Comment:
Looks fine so far, maybe just a recommendation: There's [Seq.seq](http://www.jooq.org/products/jOO%CE%BB/javadoc/0.9.11/org/jooq/lambda/Seq.html#seq-java.lang.Iterable-) and [Seq.of](http://www.jooq.org/products/jOO%CE%BB/javadoc/0.9.11/org/jooq/lambda/Seq.html#of-T...-) which I use most of the time to convert from `Iterable<T>` or `T[]` to a `Stream<T>`. A neat bonus is that we get many more useful functions with `Seq<T>`.

You may have noticed that already, preferring plain Java CL features, but I think repeating this stuff every time we want to turn an iterable into a stream is quite a mouthful.